lib/govuk_publishing_components/presenters/single_page_notification_button_helper.rb

```diff
@@ -0,0 +1,40 @@
+module GovukPublishingComponents
+  module Presenters
+    class SinglePageNotificationButtonHelper
+      attr_reader :already_subscribed, :data_attributes, :base_path, :js_enhancement, :button_type, :button_location
+
+      def initialize(local_assigns)
+        @local_assigns = local_assigns
+        @data_attributes = @local_assigns[:data_attributes] || {}
+        @js_enhancement = @local_assigns[:js_enhancement] || false
+        @already_subscribed = @local_assigns[:already_subscribed] || false
+        @base_path = @local_assigns[:base_path] || nil
```

Comment:
Some of the options are expected to be booleans, others we expect will be some other type of object (e.g strings). 
I had the boolean ones default to `false`, expected string ones default to `nil` not sure if it's a convention or something I made up though. 😄
We could have them all use `nil` and it would not make a difference, it just wouldn't _feel right_.